src/DroneSafetyApi/Services/HazardsToHeatmapResponse.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using DroneSafetyApi.Models;
+using Microsoft.Azure.Documents.Spatial;
+
+namespace DroneSafetyApi.Services
+{
+    public class HazardsToHeatmapResponse : HazardsToHeatmapsResponseNoCompositionSendAllSources
+    {
+
+        public override HeatMap ConvertToHeatmap(BoundingBox area, int width, int height, IEnumerable<Hazard> hazards)
+        {
+            // Initialise Heatmap
+            HeatMap heatmap = new HeatMap(area.Min.Longitude, area.Max.Longitude, area.Min.Latitude, area.Max.Latitude, width, height);
+
+            foreach (Hazard hazard in hazards)
```

Comment:
No we are collecting all shapes of hazards, then we are splitting them by datasource, and only then we are splitting them based on shapes. Would be messier to keep three different lists for each datasource, so we are sticking with this way